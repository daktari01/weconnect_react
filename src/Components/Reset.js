import React, { Component } from "react";
import axios from "axios";
import PropTypes from "prop-types";
import { toast } from "react-toastify";
import { Redirect } from "react-router-dom";
import NavBar from "./NavBar";
import Footer from "./Footer";
import signInLogo from "../static/img/logos/weConnect.png";
import { localApi } from "../utilities/api";

/**
 * Handle Reset component
 */
class Reset extends Component {
  state = {
    email: "",
    fireRedirect: false
  };

  componentDidMount() {
    const reset_token = this.props.match.params.token;
    this.setState({
      reset_token: reset_token
    });
  }

  /**
   * Handle onChange event for inputs
   */
  handleChange = event => {
    this.setState({
      [event.target.name]: event.target.value
    });
  };

  /**
   * Handle the submit function
   */
  handleSubmit = event => {
    event.preventDefault();
    const newPass = {
      new_password: this.state.password,
      confirm_new_password: this.state.confirmPassword
    };
    // Make POST request
    axios
      .post(localApi + "auth/reset/" + this.state.reset_token, newPass, {
        headers: {
          Accept: "application/json",
          "Content-type": "application/json"
        }
      })
      .then(() => {
        toast.success("Reset password successful. You can now login.");
        this.setState({ fireRedirect: true });
      })
      .catch(() => {
        toast.error("Reset password unsuccessful. Please try again");
      });
  };

  render() {
    const { fireRedirect } = this.state;
    return (
      <div>
        <section id="body">
          <NavBar />
          <div className="container col-min-6" id="loginContainer">
            <div className="wrapper" id="loginWrapper">
              <form>
                <div className="col-sm-12" id="loginLogo">
                  <img className="img-fluid" alt="logo" src={signInLogo} />
                </div>
                <div className="form-group">
                  <label htmlFor="loginEmail">Password</label>
                  <br />
                  <input
                    type="password"
                    className="form-control"
                    id="inputPassword"
                    placeholder="Enter password"
                    name="password"
                    value={this.state.password}
                    onChange={this.handleChange}
                  />
                </div>
                <div className="form-group">
                  <label htmlFor="loginEmail">Confirm Password</label>
                  <br />
                  <input
                    type="password"
                    className="form-control"
                    id="inputConfirmPassword"
                    placeholder="Enter confirm password"
                    name="confirmPassword"
                    value={this.state.confirmPassword}
                    onChange={this.handleChange}
                  />
                </div>
                <button
                  type="submit"
                  className="btn btn-primary btn-block"
                  onClick={this.handleSubmit}
                >
                  Reset Password
                </button>
                <br />
              </form>
            </div>
          </div>
        </section>
        <Footer />
        {fireRedirect && <Redirect to="/login" />}
      </div>
    );
  }
}
Reset.propTypes = {
  match: PropTypes.object,
};
export default Reset;
