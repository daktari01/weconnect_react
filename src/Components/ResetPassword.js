import React, { Component } from "react";
import axios from "axios";
import { toast } from "react-toastify";
import { Redirect } from "react-router-dom";
import NavBar from "./NavBar";
import Footer from "./Footer";
import signInLogo from "../static/img/logos/weConnect.png";
import { localApi } from "../utilities/api";

  /**
   * Handle ResetPassword class
   */
class ResetPassword extends Component {
  state = {
    email: ""
  };

  /**
   * Handle onChange event for inputs
   */
  handleChange = event => {
    this.setState({
      [event.target.name]: event.target.value
    });
  };

  /**
   * Handle the submit function
   */
  handleSubmit = event => {
    event.preventDefault();
    const newPass = {
      email: this.state.email
    };
    // Make POST request
    axios
      .post(localApi + "auth/reset-password", newPass, {
        headers: {
          Accept: "application/json",
          "Content-type": "application/json"
        }
      })
      .then(response => {
        console.log(response.data);
        toast.success(
          "Almost there. Go to your email address for a password reset link."
        );
        return <Redirect to="/login" />;
      })
      .catch(error => {
        toast.error("Reset password unsuccessful. Please try again");
      });
  };

  render() {
    return (
      <div>
        <section id="body">
          <NavBar />
          <div className="container col-min-6" id="loginContainer">
            <div className="wrapper" id="loginWrapper">
              <form>
                <div className="col-sm-12" id="loginLogo">
                  <img className="img-fluid" alt="logo" src={signInLogo} />
                </div>
                <div className="form-group">
                  <label htmlFor="loginEmail">Email</label>
                  <br />
                  <input
                    type="text"
                    className="form-control"
                    id="loginEmailInput"
                    placeholder="Enter email to reset password"
                    name="email"
                    value={this.state.email}
                    onChange={this.handleChange}
                  />
                </div>
                <button
                  type="submit"
                  className="btn btn-primary btn-block"
                  onClick={this.handleSubmit}
                >
                  Reset Password
                </button>
                <br />
              </form>
            </div>
          </div>
        </section>
        <Footer />
      </div>
    );
  }
}

export default ResetPassword;
