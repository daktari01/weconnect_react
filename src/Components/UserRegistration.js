import React, { Component } from "react";
import axios from "axios";
import { toast } from "react-toastify";
import { Redirect } from "react-router-dom";
import NavBar from "./NavBar";
import Footer from "./Footer";
import signInLogo from "../static/img/logos/weConnect.png";
import { localApi } from "../utilities/api";

class UserRegistration extends Component {
  state = {
    firstName: "",
    lastName: "",
    email: "",
    username: "",
    password: "",
    confirmPassword: ""
  };

  handleChange = event => {
    this.setState({
      [event.target.name]: event.target.value
    });
  };

  handleSubmit = event => {
    event.preventDefault();
    const newUser = {
      first_name: this.state.firstName,
      last_name: this.state.lastName,
      email: this.state.email,
      username: this.state.username,
      first_password: this.state.password,
      confirm_password: this.state.confirmPassword
    };
    // Make POST request
    axios
      .post(localApi + "auth/register", newUser, {
        headers: {
          Accept: "application/json",
          "Content-type": "application/json"
        }
      })
      .then(response => {
        console.log(response.data);
        toast.success("Registration successful");
        return <Redirect to="/login" />;
      })
      .catch(error => {
        toast.error("Registration unsuccessful. Please try again");
      });
  };

  render() {
    return (
      <div>
        <section id="body">
          <NavBar />
          <div className="container d-flex p-2 col-min-6" id="loginContainer">
            <div className="wrapper" id="loginWrapper">
              <form onSubmit={this.handleSubmit}>
                <div className="col-sm-12" id="loginLogo">
                  <img
                    className="img-fluid"
                    src={signInLogo}
                    alt="WeConnect logo"
                  />
                </div>
                <div className="form-group">
                  <label htmlFor="loginEmail">First name</label>
                  <input
                    type="text"
                    className="form-control"
                    id="firstNameInput"
                    name="firstName"
                    placeholder="Enter first name"
                    value={this.state.firstName}
                    onChange={this.handleChange}
                  />
                </div>
                <div className="form-group">
                  <label htmlFor="loginEmail">Last name</label>
                  <input
                    type="text"
                    className="form-control"
                    id="lastNameInput"
                    name="lastName"
                    placeholder="Enter last name"
                    value={this.state.lastName}
                    onChange={this.handleChange}
                  />
                </div>
                <div className="form-group">
                  <label htmlFor="loginEmail">Email address</label>
                  <input
                    type="email"
                    className="form-control"
                    id="emailInput"
                    name="email"
                    placeholder="Enter email"
                    value={this.state.email}
                    onChange={this.handleChange}
                  />
                </div>
                <div className="form-group">
                  <label htmlFor="loginUsername">Username</label>
                  <input
                    type="text"
                    className="form-control"
                    id="loginUsernameInput"
                    name="username"
                    placeholder="Enter username"
                    value={this.state.username}
                    onChange={this.handleChange}
                  />
                </div>
                <div className="form-group">
                  <label htmlFor="loginPassword">Password</label>
                  <input
                    type="password"
                    className="form-control"
                    id="loginPasswordInput"
                    placeholder="Enter password"
                    name="password"
                    value={this.state.password}
                    onChange={this.handleChange}
                  />
                </div>
                <div className="form-group">
                  <label htmlFor="loginPassword">Confirm Password</label>
                  <input
                    type="password"
                    className="form-control"
                    id="loginConfirmPasswordInput"
                    name="confirmPassword"
                    placeholder="Confirm your password"
                    value={this.state.confirmPassword}
                    onChange={this.handleChange}
                  />
                </div>
                <button
                  type="submit"
                  className="btn btn-primary btn-block"
                >
                  Register
                </button>
                <p>
                  Have an account? <a href="login">Login</a>
                </p>
              </form>
            </div>
          </div>
        </section>
        <Footer />
      </div>
    );
  }
}

export default UserRegistration;
