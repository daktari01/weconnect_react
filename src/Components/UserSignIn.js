import React, { Component } from "react";
import axios from "axios";
import { toast } from "react-toastify";
import { Redirect } from "react-router-dom";
import NavBar from "./NavBar";
import Footer from "./Footer";
import signInLogo from "../static/img/logos/weConnect.png";
import { localApi } from "../utilities/api";

/**
 * Class to handle user login
 */
class UserSignIn extends Component {
  state = {
    username: "",
    password: "",
    fireRedirect: false
  };

  /**
   * Handle onChange event for inputs
   */
  handleChange = event => {
    this.setState({
      [event.target.name]: event.target.value
    });
  };

  /**
   * Handle the submit function
   */
  handleSubmit = event => {
    event.preventDefault();
    const { username, password } = this.state;
    const newLogin = { username, password };
    // Make POST request
    axios
      .post(localApi + "auth/login", newLogin, {
        headers: {
          Accept: "application/json",
          "Content-type": "application/json"
        }
      })
      .then(response => {
        // appAuth.authenticate();
        toast.success("Login successful");
        localStorage.setItem("access_token", response.data.token);
        localStorage.setItem("user_id", response.data.user_id);
        localStorage.setItem("isAuthenticated", true);
        this.setState({ fireRedirect: true });
      })
      .catch(error => {
        console.log(error);
        toast.error("Login unsuccessful. Please try again");
      });
  };

  render() {
    const { from } = this.props.location.state || "/my-businesses";
    const { username, password, fireRedirect } = this.state;
    return (
      <div>
        <section id="body">
          <NavBar />
          <div className="container col-min-6" id="loginContainer">
            <div className="wrapper" id="loginWrapper">
              <form onSubmit={this.handleSubmit}>
                <div className="col-sm-12" id="loginLogo">
                  <img className="img-fluid" alt="logo" src={signInLogo} />
                </div>
                <div className="form-group">
                  <label htmlFor="loginEmail">Username</label>
                  <input
                    type="text"
                    className="form-control"
                    id="loginUsernameInput"
                    placeholder="Enter username"
                    name="username"
                    value={username}
                    onChange={this.handleChange}
                  />
                </div>
                <div className="form-group">
                  <label htmlFor="loginPassword">Password</label>
                  <input
                    type="password"
                    className="form-control"
                    id="loginPasswordInput"
                    placeholder="Password"
                    name="password"
                    value={password}
                    onChange={this.handleChange}
                  />
                </div>
                <br />
                <button
                  type="submit"
                  className="btn btn-primary btn-block"
                >
                  Login
                </button>
                <br />
                <p>
                  Do not have an account? <a href="/register">Register</a>
                </p>
                <p>
                  <a href="/reset-password">Forgot password?</a>
                </p>
              </form>
            </div>
          </div>
        </section>
        <Footer />
        {fireRedirect && <Redirect to={from || "/businesses"} />}
      </div>
    );
  }
}

export default UserSignIn;
