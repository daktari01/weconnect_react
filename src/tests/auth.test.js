import Enzyme, { shallow, render, mount } from "enzyme";
import React from "react";
import UserRegistration from "../Components/UserRegistration";
import UserSignIn from "../Components/UserSignIn";
import ResetPassword from "../Components/ResetPassword";
import Reset from "../Components/Reset";
import { shallowToJson } from "enzyme-to-json";
import sinon from "sinon";
import moxios from "moxios";

describe("UserRegistration renders correctly", () => {
  const wrapper = shallow(<UserRegistration />);
  it("Succeeds", () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it("Handles handleSubmit on submit ", () => {
    let handleSubmit = sinon.spy;
    let wrapper = mount(<UserRegistration onSubmit={handleSubmit} />);
    wrapper.find("form").simulate("submit");
    moxios.wait();
  });

  it("Handles handleChange on submit ", () => {
    let handleChange = sinon.spy;
    let wrapper = mount(<UserRegistration onChange={handleChange} />);
    wrapper.find("#firstNameInput").simulate("change");
  });
});

describe("UserSignIn renders correctly", () => {
  const test_state = {
    state: "james",
  };
  const wrapper = shallow(<UserSignIn location={test_state} />);
  it("Succeeds", () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it("Handles handleSubmit on submit ", () => {
    let handleSubmit = sinon.spy;
    let wrapper = mount(<UserSignIn location={test_state} onSubmit={handleSubmit} />);
    wrapper.find("form").simulate("submit");
    moxios.wait();
  });

  it("Handles handleChange on submit ", () => {
    let handleChange = sinon.spy;
    let wrapper = mount(<UserSignIn location={test_state} onChange={handleChange} />);
    wrapper.find("#loginUsernameInput").simulate("change");
  });
});

describe("ResetPassword renders correctly", () => {
  const wrapper = shallow(<ResetPassword />);
  it("Succeeds", () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});

describe("Reset renders correctly", () => {
  const token = {
      params: {
        token:
          "eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJ1c2VybmFtZSI6ImphbWVzIiwiZXhwIjoxNTMzNjI5NzExfQ.8mPPduU60z1G8OuXz27AUEPql0ETLQ-_HoTvif3_hbQ"
      }
  };
  const wrapper = shallow(<Reset match={token}/>);
  it("Succeeds", () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
