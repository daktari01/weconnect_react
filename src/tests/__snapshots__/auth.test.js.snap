// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reset renders correctly Succeeds 1`] = `
<div>
  <section
    id="body"
  >
    <NavBar />
    <div
      className="container col-min-6"
      id="loginContainer"
    >
      <div
        className="wrapper"
        id="loginWrapper"
      >
        <form>
          <div
            className="col-sm-12"
            id="loginLogo"
          >
            <img
              alt="logo"
              className="img-fluid"
              src="weConnect.png"
            />
          </div>
          <div
            className="form-group"
          >
            <label
              htmlFor="loginEmail"
            >
              Password
            </label>
            <br />
            <input
              className="form-control"
              id="inputPassword"
              name="password"
              onChange={[Function]}
              placeholder="Enter password"
              type="password"
            />
          </div>
          <div
            className="form-group"
          >
            <label
              htmlFor="loginEmail"
            >
              Confirm Password
            </label>
            <br />
            <input
              className="form-control"
              id="inputConfirmPassword"
              name="confirmPassword"
              onChange={[Function]}
              placeholder="Enter confirm password"
              type="password"
            />
          </div>
          <button
            className="btn btn-primary btn-block"
            onClick={[Function]}
            type="submit"
          >
            Reset Password
          </button>
          <br />
        </form>
      </div>
    </div>
  </section>
  <Footer />
</div>
`;

exports[`ResetPassword renders correctly Succeeds 1`] = `
<div>
  <section
    id="body"
  >
    <NavBar />
    <div
      className="container col-min-6"
      id="loginContainer"
    >
      <div
        className="wrapper"
        id="loginWrapper"
      >
        <form>
          <div
            className="col-sm-12"
            id="loginLogo"
          >
            <img
              alt="logo"
              className="img-fluid"
              src="weConnect.png"
            />
          </div>
          <div
            className="form-group"
          >
            <label
              htmlFor="loginEmail"
            >
              Email
            </label>
            <br />
            <input
              className="form-control"
              id="loginEmailInput"
              name="email"
              onChange={[Function]}
              placeholder="Enter email to reset password"
              type="text"
              value=""
            />
          </div>
          <button
            className="btn btn-primary btn-block"
            onClick={[Function]}
            type="submit"
          >
            Reset Password
          </button>
          <br />
        </form>
      </div>
    </div>
  </section>
  <Footer />
</div>
`;

exports[`UserRegistration renders correctly Succeeds 1`] = `
<div>
  <section
    id="body"
  >
    <NavBar />
    <div
      className="container d-flex p-2 col-min-6"
      id="loginContainer"
    >
      <div
        className="wrapper"
        id="loginWrapper"
      >
        <form
          onSubmit={[Function]}
        >
          <div
            className="col-sm-12"
            id="loginLogo"
          >
            <img
              alt="WeConnect logo"
              className="img-fluid"
              src="weConnect.png"
            />
          </div>
          <div
            className="form-group"
          >
            <label
              htmlFor="loginEmail"
            >
              First name
            </label>
            <input
              className="form-control"
              id="firstNameInput"
              name="firstName"
              onChange={[Function]}
              placeholder="Enter first name"
              type="text"
              value=""
            />
          </div>
          <div
            className="form-group"
          >
            <label
              htmlFor="loginEmail"
            >
              Last name
            </label>
            <input
              className="form-control"
              id="lastNameInput"
              name="lastName"
              onChange={[Function]}
              placeholder="Enter last name"
              type="text"
              value=""
            />
          </div>
          <div
            className="form-group"
          >
            <label
              htmlFor="loginEmail"
            >
              Email address
            </label>
            <input
              className="form-control"
              id="emailInput"
              name="email"
              onChange={[Function]}
              placeholder="Enter email"
              type="email"
              value=""
            />
          </div>
          <div
            className="form-group"
          >
            <label
              htmlFor="loginUsername"
            >
              Username
            </label>
            <input
              className="form-control"
              id="loginUsernameInput"
              name="username"
              onChange={[Function]}
              placeholder="Enter username"
              type="text"
              value=""
            />
          </div>
          <div
            className="form-group"
          >
            <label
              htmlFor="loginPassword"
            >
              Password
            </label>
            <input
              className="form-control"
              id="loginPasswordInput"
              name="password"
              onChange={[Function]}
              placeholder="Enter password"
              type="password"
              value=""
            />
          </div>
          <div
            className="form-group"
          >
            <label
              htmlFor="loginPassword"
            >
              Confirm Password
            </label>
            <input
              className="form-control"
              id="loginConfirmPasswordInput"
              name="confirmPassword"
              onChange={[Function]}
              placeholder="Confirm your password"
              type="password"
              value=""
            />
          </div>
          <button
            className="btn btn-primary btn-block"
            type="submit"
          >
            Register
          </button>
          <p>
            Have an account? 
            <a
              href="login"
            >
              Login
            </a>
          </p>
        </form>
      </div>
    </div>
  </section>
  <Footer />
</div>
`;

exports[`UserSignIn renders correctly Succeeds 1`] = `
<div>
  <section
    id="body"
  >
    <NavBar />
    <div
      className="container col-min-6"
      id="loginContainer"
    >
      <div
        className="wrapper"
        id="loginWrapper"
      >
        <form
          onSubmit={[Function]}
        >
          <div
            className="col-sm-12"
            id="loginLogo"
          >
            <img
              alt="logo"
              className="img-fluid"
              src="weConnect.png"
            />
          </div>
          <div
            className="form-group"
          >
            <label
              htmlFor="loginEmail"
            >
              Username
            </label>
            <input
              className="form-control"
              id="loginUsernameInput"
              name="username"
              onChange={[Function]}
              placeholder="Enter username"
              type="text"
              value=""
            />
          </div>
          <div
            className="form-group"
          >
            <label
              htmlFor="loginPassword"
            >
              Password
            </label>
            <input
              className="form-control"
              id="loginPasswordInput"
              name="password"
              onChange={[Function]}
              placeholder="Password"
              type="password"
              value=""
            />
          </div>
          <br />
          <button
            className="btn btn-primary btn-block"
            type="submit"
          >
            Login
          </button>
          <br />
          <p>
            Do not have an account? 
            <a
              href="/register"
            >
              Register
            </a>
          </p>
          <p>
            <a
              href="/reset-password"
            >
              Forgot password?
            </a>
          </p>
        </form>
      </div>
    </div>
  </section>
  <Footer />
</div>
`;
